scenario_player/tasks/blockchain.py

```diff
@@ -171,23 +173,97 @@ def _run(self, *args, **kwargs) -> Dict[str, Any]:  # pylint: disable=unused-arg
+        self, runner: scenario_runner.ScenarioRunner, config: Any, parent: "Task" = None
+    ) -> None:
+        config.update({"contract_name": "TokenNetwork", "event_name": "ChannelSettled"})
+        super().__init__(runner, config, parent)
+
+        verify_config(config, required_keys=["initiator", "partner", "channel_info_key"])
+
+        self.initiator = self._get_node_address(config["initiator"])
+        self.initiator_amount = config.get("initiator_amount", None)
+        self.partner = self._get_node_address(config["partner"])
+        self.partner_amount = config.get("partner_amount", None)
+
+    def _run(self, *args, **kwargs) -> Dict[str, Any]:  # pylint: disable=unused-argument
+        channel_infos = self._runner.task_storage[STORAGE_KEY_CHANNEL_INFO].get(
+            self._config["channel_info_key"]
+        )
+        if channel_infos is None:
+            raise ScenarioError(
+                f"No stored channel info found for key '{self._config['channel_info_key']}'."
             )
 
-        return {"events": events}
+        channel_identifier = channel_infos["channel_identifier"]
+        self.event_args.update(
+            {
+                "channel_identifier": channel_identifier,
+            }
+        )
+        event_dict = super()._run(*args, **kwargs)
+        events = event_dict["events"]
+
+        # query in "both directions"
+        events = self._filter_for_channel_settled(
+            events, self.initiator, self.initiator_amount, self.partner, self.partner_amount
+        )
+        if not events:
+            events = self._filter_for_channel_settled(
+                events, self.partner, self.partner_amount, self.initiator, self.initiator_amount
+            )
+        if len(events) != 1:
+            raise ScenarioAssertionError("Did not find expected ChannelSettled event!")
+        coop_settle_event = events[0]
+        transaction_hash = coop_settle_event["transactionHash"]
+        transaction = self.web3.eth.get_transaction(transaction_hash)
+        transaction_sender = transaction["from"]
+        if not transaction_sender == self.initiator:
+            raise ScenarioAssertionError(
+                f"The ChannelSettled event was emitted from a tx by {transaction_sender}, but was "
+                f"expected to be emitted from a tx by {self.initiator}"
+            )
+        return event_dict
+
+    @staticmethod
+    def _filter_for_channel_settled(
+        events, participant1, participant1_amount, participant2, participant2_amount
+    ):
+        event_args = {"participant1": participant1, "participant2": participant2}
+        if participant1_amount is not None:
+            event_args["participant1_amount"] = participant1_amount
+        if participant2_amount is not None:
+            event_args["participant2_amount"] = participant2_amount
+        event_args_items = event_args.items()
+        return [e for e in events if e["args"] and event_args_items & e["args"].items()]
```

Comment:
Do we need to return a mutable object from this function? Looking at the usage, I would say not, so I'd rather we return an immutable object, i.e. a tuple.